pkg/controller/node/node_controller.go

```diff
@@ -763,28 +785,74 @@ func (ctrl *Controller) handleErr(err error, key interface{}) {
 // 2. If a MachineConfig changes, we should wait for the OS image build to be
 // ready so we can update both the nodes' desired MachineConfig and desired
 // image annotations simultaneously.
+
+func (ctrl *Controller) GetConfigAndBuild(pool *mcfgv1.MachineConfigPool) (*mcfgv1alpha1.MachineOSConfig, *mcfgv1alpha1.MachineOSBuild, error) {
+	var ourConfig *mcfgv1alpha1.MachineOSConfig
+	var ourBuild *mcfgv1alpha1.MachineOSBuild
+	configList, err := ctrl.moscLister.List(labels.Everything())
+	if err != nil {
+		return nil, nil, err
+	}
+
+	for _, config := range configList {
+		if config.Spec.MachineConfigPool.Name == pool.Name {
+			ourConfig = config
+			break
```

Comment:
thought: What should happen if `ourConfig` / `ourBuild` is nil, meaning that there is no config or build for the given MachineConfigPool? Should we return an error in that case?